### PR TITLE
Finished register

### PR DESCRIPTION
**Pull request by dinhhieu165**

### CODE REVIEW

File: app/assets/stylesheets/custom.scss [hunk 1]
```diff
@@ -0,0 +1,43 @@
+@import "bootstrap-sprockets";
+@import "bootstrap";
+
+
```
Comment (tranchien2002) [inline app/assets/stylesheets/custom.scss]: thừa line

File: app/assets/stylesheets/users.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Users controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: http://sass-lang.com/
```
Comment (tranchien2002) [inline app/assets/stylesheets/users.scss]: xóa cmt

File: config/routes.rb [hunk 3]
```diff
@@ -1,3 +1,6 @@
 Rails.application.routes.draw do
 
+  get  '/signup',  to: 'users#new'
+  post '/signup',  to: 'users#create'
+  resources :users
 end
```
Comment (tranchien2002) [inline config/routes.rb]: thừa routing

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -0,0 +1,24 @@
+class UsersController < ApplicationController
+  def show
+    @user = User.find(params[:id])
```
Comment (chuanhtuan) [inline app/controllers/users_controller.rb]: remove ()

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  end
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
```
Comment (chuanhtuan) [inline app/controllers/users_controller.rb]: remove ()

File: app/controllers/users_controller.rb [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    if @user.save
+      flash[:success] = "Your account successfully registered !!!"
+      redirect_to @user
+    else
+      render 'new'
```
Comment (chuanhtuan) [inline app/controllers/users_controller.rb]: "

File: app/controllers/users_controller.rb [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+    end
+  end
+
+  private
+    def user_params
+      params.require(:user).permit :name, :email, :username, :password, :password_confirmation, :address, :phone_number
```
Comment (chuanhtuan) [inline app/controllers/users_controller.rb]: 80 characters per line

File: app/views/users/new.html.erb [hunk 8]
```diff
@@ -0,0 +1,33 @@
+<% provide(:title, 'Register') %>
+<h1>Register</h1>
```
Comment (chuanhtuan) [inline app/views/users/new.html.erb]: i18n

File: app/views/users/new.html.erb [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+
+<div class="row">
+  <div class="col-md-4 col-md-offset-4">
+    <%= form_for(@user, url: signup_path) do |f| %>
+      <%= render 'shared/error_messages', object: f.object %>
```
Comment (chuanhtuan) [inline app/views/users/new.html.erb]: "

File: app/views/users/new.html.erb [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+      <%= f.label :password_confirmation, "Confirmation" %>
+      <%= f.password_field :password_confirmation, class: 'form-control' %>
+
+      <%= f.label :address %>
+      <%= f.text_field :address, class: 'form-control' %>
+
+      <%= f.label :phone_number %>
+      <%= f.text_field :phone_number, class: 'form-control' %>
+
+      <%= f.submit "Register", class: "btn btn-primary" %>
```
Comment (chuanhtuan) [inline app/views/users/new.html.erb]: I18n

File: Gemfile [hunk 11]
```diff
@@ -23,6 +23,10 @@ gem "jbuilder", "~> 2.5"
 
 gem "bcrypt", "~> 3.1"
 
+gem 'bootstrap-sass', '3.3.7'
```
Comment (chuanhtuan) [inline Gemfile]: "

File: app/views/users/new.html.erb [hunk 12]
```diff
@@ -0,0 +1,33 @@
+<% provide(:title, 'Register') %>
+<h1>Register</h1>
+
+<div class="row">
+  <div class="col-md-4 col-md-offset-4">
+    <%= form_for(@user, url: signup_path) do |f| %>
+      <%= render 'shared/error_messages', object: f.object %>
+
+      <%= f.label :name %>
```
Comment (chuanhtuan) [inline app/views/users/new.html.erb]: I18n

**Pull request merged**